hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java

```diff
@@ -277,53 +246,67 @@ public static SCMSecurityProtocolClientSideTranslatorPB getScmSecurityClient(
    * @throws IllegalArgumentException if any values are not in the 'host'
    *             or host:port format.
    */
-  public static Optional<Integer> getPortNumberFromConfigKeys(
+  public static OptionalInt getPortNumberFromConfigKeys(
       Configuration conf, String... keys) {
     for (final String key : keys) {
       final String value = conf.getTrimmed(key);
-      final Optional<Integer> hostPort = getHostPort(value);
+      final OptionalInt hostPort = getHostPort(value);
       if (hostPort.isPresent()) {
         return hostPort;
       }
     }
-    return Optional.empty();
+    return OptionalInt.empty();
   }
 
   /**
    * Retrieve the socket addresses of all storage container managers.
    *
-   * @param conf
    * @return A collection of SCM addresses
    * @throws IllegalArgumentException If the configuration is invalid
    */
   public static Collection<InetSocketAddress> getSCMAddresses(
-      Configuration conf) throws IllegalArgumentException {
-    Collection<InetSocketAddress> addresses =
-        new HashSet<InetSocketAddress>();
+      Configuration conf) {
     Collection<String> names =
         conf.getTrimmedStringCollection(ScmConfigKeys.OZONE_SCM_NAMES);
-    if (names == null || names.isEmpty()) {
+    if (names.isEmpty()) {
       throw new IllegalArgumentException(ScmConfigKeys.OZONE_SCM_NAMES
           + " need to be a set of valid DNS names or IP addresses."
-          + " Null or empty address list found.");
+          + " Empty address list found.");
     }
 
-    final Optional<Integer> defaultPort = Optional
-        .of(ScmConfigKeys.OZONE_SCM_DEFAULT_PORT);
+    Collection<InetSocketAddress> addresses = new HashSet<>(names.size());
     for (String address : names) {
       Optional<String> hostname = getHostName(address);
       if (!hostname.isPresent()) {
         throw new IllegalArgumentException("Invalid hostname for SCM: "
-            + hostname);
+            + address);
       }
-      Optional<Integer> port = getHostPort(address);
-      InetSocketAddress addr = NetUtils.createSocketAddr(hostname.get(),
-          port.orElse(defaultPort.get()));
+      int port = getHostPort(address)
+          .orElse(ScmConfigKeys.OZONE_SCM_DEFAULT_PORT);
+      InetSocketAddress addr = NetUtils.createSocketAddr(hostname.get(), port);
       addresses.add(addr);
     }
     return addresses;
   }
-  
+
+  /**
+   * Same as {@link #getSCMAddresses(Configuration)}, but verifies that
+   * only one SCM address is configured, as currently multiple ones are not
+   * supported.  When multi-SCM support is added, simply replace calls with the
+   * delegate method.
+   *
+   * @return a collection of SCM addresses with exactly 1 item
+   * @throws IllegalArgumentException if {@code conf} has more than one SCM
+   *         address or it has none
+   */
+  public static Collection<InetSocketAddress> getSingleSCMAddress(
```

Comment:
should we just return a single InetSocketAddress here instead of a collection?

Reply:
Thanks @xiaoyuyao reviewing this.

> should we just return a single InetSocketAddress here instead of a collection?

I think it depends on whether we may want to keep this method when multi-SCM support is added.  I considered it to be a temporary one.  Otherwise a single address might be better, as it simplifies callers.

Reply:
There are a bunch of other SCM addresses that need to be changed anyway when SCM-HA is added. My suggestion is to leave this one as a single to be consistent with current non-HA SCM and change to collection when all the other addresses are changed for SCM-HA.

Reply:
> There are a bunch of other SCM addresses that need to be changed anyway when SCM-HA is added.

I came to the same conclusion.